### ISSUE TITLE
Deploy Multicall3 on Linea Testnet

### ISSUE THREAD

**Issue by croll83** [mds1/multicall#77]
Hello team! 

we would love to get Multicall3 deployed on Linea Testnet 

Linea is a zkEVM L2 on top of Ethereum (https://linea.build, https://docs.linea.build), currently bootstrapped by ConsenSys. 

ChainID: 59140
Network Name: Linea Goerli Testnet
RPC URL:https://rpc.goerli.linea.build
Currency Symbol: ETH
Block Explorer URL: https://explorer.goerli.linea.build

I sent some ETH on Linea to the deployer address.

Happy to integrate this issue if you need more info.

Many thanks

**Comment by mds1** [comment]
Do deployments on Linea allow for the same addresses as Ethereum mainnet? For example, polygon's zkEVM is compatible in that sense, whereas zkSync's era [is not](https://github.com/mds1/multicall/issues/76#issuecomment-1496841492), but will be eventually so I haven't deployed there yet.

I checked https://docs.linea.build/developers/ethereum-differences but couldn't find any info. I could always test this myself using a different account if needed. If the addresses are computed the same, funding the deployer address would be appreciated!

**Comment by croll83** [comment]
Yes, Linea should be compatible as well. 4337 Single Entry Point is already deployed and it has the same address of all other EVM chains.

the addresses are computed the same.

I funded the deployer with 1ETH directly on Linea (already bridged from Goerli). You can simply use https://rpc.goerli.linea.build for the RPC connection.

we don’t have etherscan yet, but if you need to verify the contract you can use truffle or hardhat plugin to do that. I can support in configuring the plugin to verify the code (if needed).

many thanks

**Comment by mds1** [comment]
Done, thanks! https://explorer.goerli.linea.build/address/0xcA11bde05977b3631167028862bE2a173976CA11/contracts#address-tabs
